jax/numpy/lax_numpy.py

```diff
@@ -545,6 +545,28 @@ def exp2(x):
   return lax.exp(lax.mul(lax.log(_constant_like(x, 2)), x))
 
 
+@_wraps(onp.signbit)
+def signbit(x):
```

Comment:
An observation: if it weren't for NaNs, we could probably just call `lax.sign` and post-process the output.

Reply:
Thank you for the suggestion! I roughly try something like this:

```python
@_wraps(onp.signbit)
def signbit_v2(x):
  x, = _promote_shapes("signbit", x)

  y = lax.sign(x)
  return where(lax.bitwise_or(isnan(y), y >= _constant_like(y, 0)), False, True)
```
And perf the performance
```python
In [4]: x = np.random.normal(size=(100, 100), scale=10, loc=0.0)

In [5]: %timeit jnp.signbit(x)
307 us +- 6.72 us per loop (mean +- std. dev. of 7 runs, 1000 loops each)

In [6]: %timeit jnp.signbit_v2(x)
961 us +- 24 us per loop (mean +- std. dev. of 7 runs, 1000 loops each)

In [7]: %timeit np.signbit(x)
5.75 us +- 252 ns per loop (mean +- std. dev. of 7 runs, 100000 loops each)
```
Not sure if there is any improvement that I could do.

Reply:
In general, JAX op-by-op dispatch times are higher than NumPy. We can and will lower them over time, but mostly it doesn't matter as much because you can form larger computations using `jit`.

In fact, you could probably improve both of the timings here by using a `jit` decorator on your implementations of `signbit`.